app/controllers/recipes_controller.rb

```diff
@@ -13,14 +15,41 @@ def index
 
   def show
     @recipe = Recipe.find_by_id(params[:id])
+    @recipe_foods = @recipe&.recipe_foods
     @recipe = 'No recipes' if @recipe.nil?
     @current_user = current_user
   end
 
   def destroy
     @recipe = Recipe.find_by_id(params[:id])
-    @recipe.destroy
-    flash.now[:success] = 'recipe deleted.'
-    redirect_to recipes_index_path
+    if @recipe.user == current_user
+      if @recipe.destroy
+        flash[:success] = 'recipe deleted.'
+      else
+        flash[:fail] = 'recipe deletion unsucessful.'
+      end
+    end
+    redirect_to recipes_path
+  end
+
+  def fetch_recipe
+    @recipe = Recipe.find_by_id(params[:recipe_id])
+  end
```

Comment:
[OPTIONAL] I think this function could be a private function. Please correct me if I'm wrong.